src/target/riscv/riscv.c

```diff
@@ -4565,12 +4570,8 @@ static void riscv_invalidate_register_cache(struct target *target)
 	if (!target->reg_cache)
 		return;
 
-	LOG_DEBUG("[%d]", target->coreid);
+	LOG_TARGET_DEBUG(target, "Invalidating register cache of hart: [%d]", target->coreid);
 	register_cache_invalidate(target->reg_cache);
-	for (size_t i = 0; i < target->reg_cache->num_regs; ++i) {
-		struct reg *reg = &target->reg_cache->reg_list[i];
-		reg->valid = false;
-	}
```

Comment:
This looks right, but I worry there is some subtle reason this code exists. What testing have you done? I think `make all` in riscv-tests/debug should be good enough. (Some of the `multispike` tests may fail. That's OK.)

Reply:
I tested it manually on spike via telnet console. I will tell you once I have ran the tests.

Reply:
Ok, so I can the tests, the only test which does not pass is the IcountTest (for all configurations)

Also I took a look into the original commit which introduced setting reg->valid to false and it was part of a larger commit implementing debug specification 0.13. What we see here is the leftovers after 6 years of refactoring.

Reply:
That sounds like your spike doesn't implement icount triggers, which is fine for this change.

I agree that this is probably just left-over code.